lifecycle-operator/cmd/config/config_test.go

```diff
@@ -1,13 +1,82 @@
 package config
 
 import (
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
 func TestConfigProvider(t *testing.T) {
-	provider := NewKubeConfigProvider()
+	t.Run("NewKubeConfigProvider", func(t *testing.T) {
+		provider := NewKubeConfigProvider()
+		assert.NotNil(t, provider)
+	})
 
-	assert.NotNil(t, provider)
+	t.Run("GetConfigSuccess", func(t *testing.T) {
+
+		tempDir := setupMockedKubeConfig(t)
+		defer os.RemoveAll(tempDir)
+
+		provider := NewKubeConfigProvider()
+
+		config, err := provider.GetConfig()
+		assert.NoError(t, err)
+		assert.NotNil(t, config)
+	})
+}
+
+func setupMockedKubeConfig(t *testing.T) string {
+	tempDir := t.TempDir()
+
+	kubeConfigFile := filepath.Join(tempDir, ".kube", "config")
+	err := os.MkdirAll(filepath.Dir(kubeConfigFile), 0755)
+	assert.NoError(t, err)
+
+	configContent := genKubeconfig("test-context")
```

Comment:
instead of a big function here you can have a const at the top of the file with the content you want in it, for instance ```
const file= \`
my very nice
long 
file
\`
```

Reply:
Alright